src/main/java/org/folio/service/storage/FolioS3ClientFactory.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.service.storage;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rest.utils.OkapiConnectionParams;
+import org.folio.s3.client.FolioS3Client;
+import org.folio.s3.client.S3ClientFactory;
+import org.folio.s3.client.S3ClientProperties;
+
+import java.util.Map;
+
+public class FolioS3ClientFactory {
+  private FolioS3Client folioS3Client;
+
+  public FolioS3ClientFactory() {
+    this.folioS3Client = null;
+  }
+
+  public FolioS3Client getFolioS3Client(Map<String, String> okapiHeaders) {
```

Comment:
I can see the bucket name is determined from okapi headers when createFolioS3Client method is called. But, The getFolioS3Client is returning the existing folioS3Client if folioS3Client is not null.  My question is, does this code works correctly in case of multiple tenants calling getFolioS3Client method. Please check

Reply:
fixed